sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -134,7 +135,6 @@ class QueryExecutionErrorsSuite extends QueryTest with SharedSparkSession {
         .collect()
     }
     assert(e2.getMessage === "The feature is not supported: " +
-      "literal for '[dotnet,Dummies]' of class " +
-      "org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema.")
+      "a type '[dotnet,Dummies]' is not supported as a literal type.")
```

Comment:
The message still can confuse users because there are no explicit literals in the code above. It would be nice if you catch the exception in the pivot code, and raise more clear exception with more precise error message.

Reply:
Thanks for the review. sorry I misunderstood and it makes sense. I’ll work on it.